### DISCUSSION TITLE
ERROR	No exports found in module "../src/main.js".

### DISCUSSION THREAD

**Discussion by NicmeisteR** [vercel/community#1910] [Help] [answered]
### Summary

Did Vercel have a breaking update since 19 Jan or something? Using my exact same source code even unchanged fails when I deploy it and I now get the following two error messages and 502's all over.

```
2023-03-29T12:17:45.647Z	9b9758ee-a50b-4ad4-8d62-eb52c5064046	ERROR	No exports found in module "../src/main.js".
2023-03-29T12:17:45.647Z	9b9758ee-a50b-4ad4-8d62-eb52c5064046	ERROR	Did you forget to export a function or a server?
RequestId: 6918c151-0e61-477b-9c1b-e05fb68d8af5 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

When I do an instant rollback everything works fine. But just to test that it wasn't my code that broke something I even reverted everything but it's still failing. Only works on a rollback which in itself is not ideal either.

### Additional information

This is a NestJS application.

**Comment by gabriel-gn** [comment]
+1 on this! I am also experiencing issues from nest.js serverless application recently, but "instant rollback" works fine

**Reply by NicmeisteR** [reply]
Glad it's not just me, hopefully can get a response from someone soon. Wasted way too much time trying to get my stuff working again haha.

**Comment by amyegan** [comment]
It's possible that a package update, either Nest.js or a different dependency, may cause this kind of problem. Have you tried pinning exact [version numbers](https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept) before deploying?

**Reply by NicmeisteR** [reply]
Yup, As mentioned I've changed nothing. I even reverted all of my changes and tried to do a deploy with the exact code/packages as the previous successful version and it still won't work. Just works when I do a rollback so I'm assuming something on Vercel's configs etc might've changed.

**Reply by amyegan** [reply]
Sorry, I didn't explain myself very well! I didn't mean that you had updated any packages, since you already said you said your source code was unchanged. I was referring to the problem where `@nestjs/core: ^9.2.1` means any minor and patch releases after `9.2.1` could be used for new installations. In my experience, this is a common cause of build problems that seem unrelated to site code.

If you remove `node_modules`, run `npm install` to get fresh copies of your dependencies, and then try to build/run locally, do you get similar errors?

If you can share a public repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), someone here may recognize the problem and have a solution. Tough to decide what else could be causing it without more information

**Reply by NicmeisteR** [reply]
I'll give it a try today and see what happens, If not I'll try and make a MRE though I'm not certain if I'll be able to reproduce it with an MRE. Thanks for the feedback. I'll revert back soon.

**Reply by NicmeisteR** [reply]
@amyegan for some reason updating my tsconfig solved the issue. Don't know why as it shouldn't have been an issue but was following this thread.
https://github.com/orgs/vercel/discussions/1911#discussioncomment-5498624

**Reply by amyegan** [reply]
I'm glad you found a solution. 😄 Thanks for coming back to share this!

**Comment by NicmeisteR** [comment, marked as answer]
The solution is in this thread:
https://github.com/orgs/vercel/discussions/1911#discussioncomment-5498624
